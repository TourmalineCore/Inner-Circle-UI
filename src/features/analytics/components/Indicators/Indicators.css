.indicators {
  display: flex;
  align-items: flex-start;
  flex-wrap: wrap;
  justify-content: space-between;
  font-family: Montserrat, sans-serif;
}

.indicators > div:nth-child(-n+2) {
  margin-right: 32px;
}
