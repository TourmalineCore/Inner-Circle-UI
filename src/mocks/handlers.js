import { rest } from 'msw';
import { API_ROOT, LINK_TO_SALARY_SERVICE } from '../common/config/config';

import totalFinanceData from './data/total-finance-data.json';
import getAnalyticsData from './data/get-analytics-data.json';

export const handlers = [
  // Handles a POST /login request
  rest.post(`${API_ROOT}${LINK_TO_SALARY_SERVICE}employees/create`, (req, res, ctx) => res(
    ctx.status(200),
  )),
  // Handles a GET /user request
  rest.get(`${API_ROOT}${LINK_TO_SALARY_SERVICE}finance/get-total-finance`, (req, res, ctx) => res(
    ctx.status(200),
    ctx.json(totalFinanceData),
  )),

  rest.get(`${API_ROOT}${LINK_TO_SALARY_SERVICE}finance/get-analytic`, (req, res, ctx) => res(
    ctx.status(200),
    ctx.json(getAnalyticsData),
  )),
];
