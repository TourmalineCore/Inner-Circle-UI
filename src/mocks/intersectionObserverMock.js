global.IntersectionObserver = class IntersectionObserver {
  // eslint-disable-next-line @typescript-eslint/no-useless-constructor, @typescript-eslint/no-empty-function
  constructor() {}

  observe() {
    return null;
  }

  disconnect() {
    return null;
  }

  unobserve() {
    return null;
  }
};
