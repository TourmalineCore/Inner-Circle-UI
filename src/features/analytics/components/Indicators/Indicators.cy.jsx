import Indicators from "./Indicators"

import { MemoryRouter } from 'react-router-dom';

import jsonMockRequest from '../../../../../cypress/fixtures/total-finance-data.json';

describe('Indicators', () => {
  beforeEach(() => {
    mockGetTotalFinanceResponse()
    mountIndicators();
    waitResponse();
  })

  it('validate right parse request for value reserveForQuarter', () => {
    cy.findByTestId("reserve-for-quarter-label").should('contain.text', jsonMockRequest.reserveFinance.reserveForQuarter)
  });

  it('validate right parse request for value reserveForHalfYear', () => {
    cy.findByTestId("reserve-for-half-year-label").should('contain.text', jsonMockRequest.reserveFinance.reserveForHalfYear)
  });

  it('validate right parse request for value reserveForYear', () => {
    cy.findByTestId("reserve-for-year-label").should('contain.text', jsonMockRequest.reserveFinance.reserveForYear)
    
  });

  it('validate right parse request for value desiredIncome', () => {
    cy.findByTestId("desired-income-label").should('contain.text', jsonMockRequest.desiredFinancialMetrics.desiredIncome)
    
  });

  it('validate right parse request for value ', () => {
    cy.findByTestId("desired-profit-label").should('contain.text', jsonMockRequest.desiredFinancialMetrics.desiredProfit)
    
  });

  it('validate right parse request for value desiredProfitability', () => {
    cy.findByTestId("desired-profitability-label").should('contain.text', jsonMockRequest.desiredFinancialMetrics.desiredProfitability)
    
  });

  it('validate right parse request for value totalExpense', () => {
    cy.findByTestId("total-expense-label").should('contain.text', jsonMockRequest.totalExpenses.totalExpense)
    
  });

  it('validate right parse request for value payrollExpense', () => {
    cy.findByTestId("payroll-expense-label").should('contain.text', jsonMockRequest.totalExpenses.payrollExpense)
    
  });

  it('validate right parse request for value officeExpense', () => {
    cy.findByTestId("office-expense-label").should('contain.text', jsonMockRequest.totalExpenses.officeExpense)
    
  });

  it('validate right parse request for value workingDaysInYear', () => {
    cy.findByTestId("working-days-in-year-label").should('contain.text', jsonMockRequest.workingDays.workingDaysInYear)
    
  });

  it('validate right parse request for value workingDaysInYearWithoutVacation', () => {
    cy.findByTestId("working-days-in-year-without-vacation-label").should('contain.text', jsonMockRequest.workingDays.workingDaysInYearWithoutVacation)
    
  });

  it('validate right parse request for value workingDaysInYearWithoutVacationAndSick', () => {
    cy.findByTestId("working-days-in-year-without-vacation-and-sick-label").should('contain.text', jsonMockRequest.workingDays.workingDaysInYearWithoutVacationAndSick)
    
  });

  it('validate right parse request for value workingDaysInMonth', () => {
    cy.findByTestId("working-days-in-month-label").should('contain.text', jsonMockRequest.workingDays.workingDaysInMonth)
    
  });

  it('validate right parse request for value workingHoursInMonth', () => {
    cy.findByTestId("working-hours-in-month-label").should('contain.text', jsonMockRequest.workingDays.workingHoursInMonth)
    
  });
  it('validate right parse request for value districtCoefficient', () => {
    cy.findByTestId("district-coefficient-label").should('contain.text', jsonMockRequest.districtCoefficient * 100)
    
  });

  it('validate right parse request for value incomeTaxPercent', () => {
    cy.findByTestId("income-tax-percent-label").should('contain.text', jsonMockRequest.incomeTaxPercent * 100)
  });

  it('validate right parse request for value minimumWage', () => {
    const convertedValue = jsonMockRequest.minimumWage.toString().replace('.', ',');
    cy.findByTestId("minimum-wage-label").should('contain.text', convertedValue)
  });
})

function mountIndicators() {
  cy.mount(
    <MemoryRouter>
      <Indicators />
    </MemoryRouter>,
  );
}

function mockGetTotalFinanceResponse() {
  cy.intercept('GET', '/api/salary/finance/get-total-finance', {
    statusCode: 200,
    fixture: 'total-finance-data.json',
  })
  .as('getTotalFinance');
}

function waitResponse() {
  cy.wait('@getTotalFinance');
}