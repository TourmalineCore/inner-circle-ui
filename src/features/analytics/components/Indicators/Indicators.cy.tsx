import Indicators from './Indicators';

describe('Indicators', () => {
  it('mount component', () => {
    cy.mount(<Indicators />);
  });
});
