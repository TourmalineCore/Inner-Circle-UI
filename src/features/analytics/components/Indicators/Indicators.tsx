import { useEffect, useState } from 'react';
import { api } from '../../../../common/api';
import ContentCard from '../../../../components/ContentCard/ContentCard';
import DefaultCardHeader from '../../../../components/DefaultCardHeader/DefaultCardHeader';
import { IndicatorsType } from '../../types';
import { formatMoney } from '../../../../common/utils/formatMoney';

import IndicatorComponent from '../IndicatorComponent/IndicatorComponent';
import IndicatorValue from '../IndicatorValue/IndicatorValue';

import './Indicators.css';

function Indicators() {
  const [indicators, setIndicators] = useState<IndicatorsType>({
    totalExpenses: {
      payrollExpense: 0,
      officeExpense: 0,
      totalExpense: 0,
    },
    desiredFinancialMetrics: {
      desiredIncome: 0,
      desiredProfit: 0,
      desiredProfitability: 0,
    },
    reserveFinance: {
      reserveForQuarter: 0,
      reserveForHalfYear: 0,
      reserveForYear: 0,
    },
    workingDays: {
      workingDaysInYear: 0,
      workingDaysInYearWithoutVacation: 0,
      workingDaysInYearWithoutVacationAndSick: 0,
      workingDaysInMonth: 0,
      workingHoursInMonth: 0,
    },
    incomeTaxPercent: 0,
    districtCoefficient: 0,
    minimumWage: 0,
  });

  useEffect(() => {
    getTotalIndicatorsAsync();
  }, []);

  return (
    <ContentCard
      isStickyHead
      headerContent={(
        <DefaultCardHeader>Indicators</DefaultCardHeader>
      )}
    >
      <div className="indicators">

        <IndicatorComponent head="Reserve for the">
          <IndicatorValue label={<li>Quarter</li>} value={formatMoney(indicators.reserveFinance.reserveForQuarter)} />
          <IndicatorValue label={<li>Half a year</li>} value={formatMoney(indicators.reserveFinance.reserveForHalfYear)} />
          <IndicatorValue label={<li>Year</li>} value={formatMoney(indicators.reserveFinance.reserveForYear)} />
        </IndicatorComponent>

        <IndicatorComponent head="Desired">
          <IndicatorValue label={<li>Income</li>} value={formatMoney(indicators.desiredFinancialMetrics.desiredIncome)} />
          <IndicatorValue label={<li>Profit</li>} value={formatMoney(indicators.desiredFinancialMetrics.desiredProfit)} />
          <IndicatorValue label={<li>Profitability</li>} value={formatMoney(indicators.desiredFinancialMetrics.desiredProfitability)} />
        </IndicatorComponent>

        <IndicatorComponent>
          <IndicatorValue label="Working days per year:" value={indicators.workingDays.workingDaysInYear.toString()} />
          <IndicatorValue label={<li>Vacation (days):</li>} value={indicators.workingDays.workingDaysInYearWithoutVacation.toString()} />
          <IndicatorValue label={<li>Sick leave (days):</li>} value={indicators.workingDays.workingDaysInYearWithoutVacationAndSick.toString()} />
          <IndicatorValue label="Working days per month (avg.):" value={indicators.workingDays.workingDaysInMonth.toString()} />
          <IndicatorValue label="Hours per month (avg.):" value={indicators.workingDays.workingHoursInMonth.toString()} />
        </IndicatorComponent>

        <IndicatorComponent>
          <IndicatorValue isColumn label="Total expense (month)" value={formatMoney(indicators.totalExpenses.totalExpense)} />
          <div className="indicators-absolute__components" style={{ display: 'flex' }}>
            <IndicatorValue isColumn label="Salary" value={formatMoney(indicators.totalExpenses.payrollExpense)} />
            <IndicatorValue isColumn label="Office" value={formatMoney(indicators.totalExpenses.officeExpense)} />
          </div>
        </IndicatorComponent>

        <div className="indicators-double__components">
          <IndicatorComponent isLite>
            <IndicatorValue isColumn label="District coefficient" value={`${indicators.districtCoefficient * 100} %`} />
          </IndicatorComponent>
          <IndicatorComponent isLite>
            <IndicatorValue isColumn label="Income tax" value={`${indicators.incomeTaxPercent * 100} %`} />
          </IndicatorComponent>
        </div>

        <IndicatorComponent isLite>
          <IndicatorValue isColumn label="Minimum Wage" value={formatMoney(indicators.minimumWage)} />
        </IndicatorComponent>

      </div>
    </ContentCard>
  );

  async function getTotalIndicatorsAsync() {
    const { data } = await api.get<IndicatorsType>('finance/get-total-finance');
    setIndicators(data);
  }
}

export default Indicators;
