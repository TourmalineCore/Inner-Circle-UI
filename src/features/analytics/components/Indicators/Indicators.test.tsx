/* eslint-disable jest/no-disabled-tests */
import { create, act } from 'react-test-renderer';
import Indicators from './Indicators';
import { server } from '../../../../mocks/server';

describe('Indicator tests', () => {
  beforeAll(() => server.listen());

  afterEach(() => server.resetHandlers());

  afterAll(() => server.close());

  it.skip('should successfully render and match to snapshot', () => {
    const component = create(<Indicators />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should render with get request with rights values and match to snapshot', async () => {
    let component;
    act(() => {
      component = create(<Indicators />);
    });

    await expect(component).toMatchSnapshot();
  });
});
