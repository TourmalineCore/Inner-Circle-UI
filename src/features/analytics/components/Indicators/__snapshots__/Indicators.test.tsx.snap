// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator tests should render with get request with rights values and match to snapshot 1`] = `null`;

exports[`Indicator tests should successfully render and match to snapshot 1`] = `
<div
  className="content-card "
  style={{}}
>
  <div
    className="content-card__sentinel content-card__sentinel--top"
  />
  <div
    className="content-card__header content-card__header--sticky"
  >
    <div
      className="default-card-header"
    >
      Indicators
    </div>
  </div>
  <div
    className="content-card__body"
  >
    <div
      className="indicators"
    >
      <div
        className="indicator-component"
      >
        <div
          className="indicator-component__head"
        >
          Reserve for the
        </div>
        <div
          className="indicator-component__values"
        >
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Quarter
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Half a year
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Year
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
        </div>
      </div>
      <div
        className="indicator-component"
      >
        <div
          className="indicator-component__head"
        >
          Desired
        </div>
        <div
          className="indicator-component__values"
        >
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Income
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Profit
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Profitability
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
        </div>
      </div>
      <div
        className="indicator-component"
      >
        <div
          className="indicator-component__values"
        >
          <div
            className="indicator-value direction-column"
          >
            <div
              className="indicator-value__label"
            >
              Total expense (month)
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
          <div
            className="indicators-absolute__components"
          >
            <div
              className="indicator-value direction-column"
            >
              <div
                className="indicator-value__label"
              >
                Salary
              </div>
              <div
                className="indicator-value__info"
              >
                0 ₽
              </div>
            </div>
            <div
              className="indicator-value direction-column"
            >
              <div
                className="indicator-value__label"
              >
                Office
              </div>
              <div
                className="indicator-value__info"
              >
                0 ₽
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="indicator-component"
      >
        <div
          className="indicator-component__values"
        >
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              Working days per year:
            </div>
            <div
              className="indicator-value__info"
            >
              0
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Vacation (days):
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              <li>
                Sick leave (days):
              </li>
            </div>
            <div
              className="indicator-value__info"
            >
              0
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              Working days per month (avg.):
            </div>
            <div
              className="indicator-value__info"
            >
              0
            </div>
          </div>
          <div
            className="indicator-value"
          >
            <div
              className="indicator-value__label"
            >
              Hours per month (avg.):
            </div>
            <div
              className="indicator-value__info"
            >
              0
            </div>
          </div>
        </div>
      </div>
      <div
        className="indicators-double__components"
      >
        <div
          className="indicator-component lite-box"
        >
          <div
            className="indicator-component__values"
          >
            <div
              className="indicator-value direction-column"
            >
              <div
                className="indicator-value__label"
              >
                District coefficient
              </div>
              <div
                className="indicator-value__info"
              >
                0 %
              </div>
            </div>
          </div>
        </div>
        <div
          className="indicator-component lite-box"
        >
          <div
            className="indicator-component__values"
          >
            <div
              className="indicator-value direction-column"
            >
              <div
                className="indicator-value__label"
              >
                Income tax
              </div>
              <div
                className="indicator-value__info"
              >
                0 %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="indicator-component lite-box"
        style={
          {
            "alignSelf": "baseline",
          }
        }
      >
        <div
          className="indicator-component__values"
        >
          <div
            className="indicator-value direction-column"
          >
            <div
              className="indicator-value__label"
            >
              Minimum Wage
            </div>
            <div
              className="indicator-value__info"
            >
              0 ₽
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="content-card__sentinel content-card__sentinel--bottom"
  />
</div>
`;
